sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -45,7 +46,10 @@ namespace Azure { namespace Core { namespace Http {
      * @throw if \p headerName is invalid.
      */
     void InsertHeaderWithValidation(
-        std::map<std::string, std::string>& headers,
+        std::map<
+            std::string,
+            std::string,
+            Azure::Core::Internal::Strings::CaseInsensitiveComparator>& headers,
```

Comment:
You wanna use a typedef for this type? because we're gonna use it in a lot of places.

like

```
using CaseInsensitiveMap = std::map<std::string, std::string, Azure::Core::Internal::Strings::CaseInsensitiveComparator>;
```